### ISSUE TITLE
Versioncmp from config.pp works erratic

### ISSUE THREAD

**Issue by marcinbojko** [chocolatey/puppet-chocolatey#134]
Related to #133 

Tested with choco 0.9.10.1/0.9.10.2/0.9.10.3
Despite choco_version being updated to the latest, versioncmp from config.pp returns values other than 0.
Next 'exec' with 'unless' is running on every puppet agent cycle because of wrong separator.

`Debug: Executing 'cmd.exe /c C:\ProgramData\chocolatey\bin\choco.exe feature list -r | findstr /B /I /C:"autoUninstaller - [enabled]"'`

**Comment by KosherSalt** [comment]
With Chocolatey v0.9.9.11 it also runs the incorrect exec, but it looks like it's trying to find use the findstr for the 0.9.10.x versions.

``` puppet
Debug: Exec[chocolatey_autouninstaller_enable](provider=windows): Executing check 'cmd.exe /c C:\ProgramData\chocolatey\bin\choco.exe feature list -r | findstr /B /I /C:"autoUninstaller|enabled|"'

Debug: Executing 'cmd.exe /c C:\ProgramData\chocolatey\bin\choco.exe feature list -r | findstr /B /I /C:"autoUninstaller|enabled|"'

Debug: Exec[chocolatey_autouninstaller_enable](provider=windows): Executing 'C:\ProgramData\chocolatey\bin\choco.exe feature -r enable -n autoUninstaller'

Debug: Executing 'C:\ProgramData\chocolatey\bin\choco.exe feature -r enable -n autoUninstaller'
Notice: /Stage[main]/Chocolatey::Config/Exec[chocolatey_autouninstaller_enable]/returns: executed successfully
```

**Comment by ferventcoder** [comment]
Hmm, perhaps the logic is flipped?

**Comment by marcinbojko** [comment]
I think not. I have almost 50 machines. Most of them are working fine, but few of them are stubborn.
Tested with notify, looks like server machines have this problem more often than others.

**Comment by ferventcoder** [comment]
No, the logic is wrong. I just looked closer. The version comes from a fact. If the version is 0, we should do nothing. Only do the secondary check when there is a 0.9.9.x version.

**Comment by ferventcoder** [comment]
Remember that facts go up to the server and come back down. Guess what that fact evaluates to on first install? 0, because it has just been provided up to the master for compilation before there is even a Chocolatey version. 

If you are still seeing this after a few runs, I think there is another issue at play, possibly in how it gathers the version. This has been fixed over in puppetlabs-chocolatey, but puppetlabs-chocolatey doesn't yet contain these fixes.

**Comment by marcinbojko** [comment]
Sorry Rob, I don't know much about puppet language. Just confirmed that all new host added today, with symptoms like in a subject are WIndows Server 2012 R2 hosts.

facter -p shows:

```
$ facter -p
architecture => x64
choco_install_path => C:\ProgramData\chocolatey
chocolateyversion => 0.9.10.3
```

**Comment by marcinbojko** [comment]
There is no change after first 200 runs ;)

**Comment by ferventcoder** [comment]
Okay. So definitely a continued bug then. :/

**Comment by marcinbojko** [comment]
Rob, I feel stupid :(
After upgrading chocolatey module to 1.2.5, according to logic in config.pp - everything works as expected. I am terribly sorry for overlooking such important step in debugging.

**Comment by ferventcoder** [comment]
@KosherSalt are you still seeing this with 1.2.5?

**Comment by ferventcoder** [comment]
Going to leave this open as it could still be an issue.

**Comment by KosherSalt** [comment]
Yes, I'm still seeing this with 1.2.5 and chocolatey version 0.9.9.11.
I don't see the issue on servers with 0.9.10.2 though.

**Comment by ferventcoder** [comment]
@KosherSalt I just released 1.2.6 that should handle this.
